### ISSUE TITLE
Feature: Add a GitHub Actions workflow for documentation site deployment. 

### ISSUE THREAD

**Issue by prjctimg** [prjctimg/huetiful#79]
This issue came from [this](https://github.com/prjctimg/huetiful-docs/pull/3#issuecomment-1794193179) pull request which failed to pass build tests. It would be nice to have a workflow that logs the deployment results where everyone can see them (in the Actions tab)

**Comment by prjctimg** [comment]
So I've found this tool called tsdoc-markdown which can generate documentation from Typescript source files. 

The tool also has a workflow file to automate the build when we do a PR. 

The docs can then be pushed to the docs repository therefore triggering a redeployment on the site. 

 [Here's the repository](https://github.com/peterpeterparker/tsdoc-markdown)

@takanome-dev Any idea how we can make such a workflow?

**Comment by takanome-dev** [comment]
> @takanome-dev Any idea how we can make such a workflow?
> 
First, try it locally and make a script to generate the docs.
Then, add a new job after `build` in [development/yml](https://github.com/prjctimg/huetiful-docs/blob/main/.github/workflows/development.yml):

```yml
docs:
    runs-on: ubuntu-20.04

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Install dependencies
        run: npm ci

      - name: Generate docs
        run: npm run docs

      - name: Commit docs
        uses: EndBug/add-and-commit@v9
        # We don't want to commit documentation changes to main
        if: ${{ github.ref != 'refs/heads/main' }}
        with:
          add: .
          default_author: github_actions
          message: '🤖 Documentation auto-update'
```

**Comment by prjctimg** [comment]
takanome-dev I have been reading around and came across "workflow triggers" .

Would it be easier to move the documentation generation to the docs repo ?

**Comment by takanome-dev** [comment]
From tsdoc-markdown
> Generates markdown documentation from TypeScript source code. Useful for generating docs from code and injecting automatically the outcome into project README files.

Will we be able to generate docs if there are no ts files?
Also, why separate the docs from this repo?

**Comment by prjctimg** [comment]
Good question @takanome-dev 

I've been considering this as well given the obvious disadvantages I'm seeing of seperating the two right now.


I had split the docs because I was failing to configure nextjs to share the repo alongside my code. 

If you're suggesting that we unite the docs and code together I'm 💯 with you.

**Comment by prjctimg** [comment]
Are you suggesting we use GitHub pages instead ?

**Comment by takanome-dev** [comment]
> Are you suggesting we use GitHub pages instead ?

No, we can totally keep using Nextjs and vercel

> I had split the docs because I was failing to configure nextjs to share the repo alongside my code.

I see, have you considered setting up monorepo (using turborepo or nx)?
There is a bit of a learning curve but it's worth it. Pick one and read the docs. Try to set up a new monorepo project (private if you want) and play with it to understand how it works.

Then, you can try, in a private repo, creating a monorepo containing this repo and the docs (maybe also add an example folder where you can add a simple project that uses our package) and also try setting up the gh actions.

That's a lot :sweat_smile: You can skip the last paragraph and directly create a PR where you move this project to a monorepo. Of course, I'll be more than happy to help with that. Although I'm not an expert in setting up monorepos or anything else :sweat_smile: but we can all learn from each other and most importantly by making mistakes :v:

**Comment by prjctimg** [comment]
No worries! My network📡 is not at its peak either

In the meantime I've cloned the docs repo inside the project folder and trying to add support for GitHub Flavoured Markdown and p5js.

Committing recent changes

**Comment by takanome-dev** [comment]
Ahh man, I'm so sorry :sweat: . Forgot this again. If someone is willing to give it a try, feel free to take it. Not sure when I'll be able to do this

**Comment by foss2cyber** [comment]
I would've taken this up, but I've been  held up by some.....stuff. Will try to get back to this asap. @prjctimg @takanome-dev

**Comment by prjctimg** [comment]
> Ahh man, I'm so sorry 😓 . Forgot this again. If someone is willing to give it a try, feel free to take it. Not sure when I'll be able to do this

No worries!

Currently working on a lot of changes and normalizing behaviour across conversions. This December I won't release anything until everything is stabilized.


You can always contribute at a more convenient time 👍


Happy holidays, devs!
